orchestrator/src/server.ts

```diff
@@ -89,115 +89,99 @@ async function runPreprocessorsParallel(data: Record<string, unknown>, preproces
     if (data["preprocessors"] === undefined) {
         data["preprocessors"] = {};
     }
+
     let currentPriorityGroup: number | undefined = undefined;
     let promises: Promise<Response | void>[] = [];
 
+    // handles only the promises in the current priority group & resets promises after processsing each group.
     const awaitResponses = async () => {
-        const responses = (await Promise.all(promises)).filter(a => a instanceof Response) as Response[];
-        for (const resp of responses) {
-            // const resp = await promise;
-            // OK data returned
-            if (resp.status === 200) {
-                try {
-                    const json = await resp.json();
-                    if (ajv.validate("https://image.a11y.mcgill.ca/preprocessor-response.schema.json", json)) {
-                        (data["preprocessors"] as Record<string, unknown>)[json["name"]] = json["data"];
-                    } else {
-                        console.error("Preprocessor response failed validation!");
-                        console.error(JSON.stringify(ajv.errors));
-                    }
-                } catch (err) {
-                    console.error("Error occurred on fetch from " + resp.url);
-                    console.error(err);
-                }
-            // No Content preprocessor not applicable
-            } else if (resp.status === 204) {
-                continue;
-            } else {
-                try {
-                    const result = await resp.json();
-                    throw result;
-                } catch (err) {
-                    console.error("Error occurred on fetch from " + resp.url);
-                    console.error(err);
-                }
-            }
-        }
+        await Promise.all(promises);
         promises = [];
```

Comment:
You've moved all the logic for this function out of the function, so it no longer serves any purpose. I can't see the point in keeping it.